Userland/Libraries/LibJS/Runtime/AbstractOperations.cpp

```diff
@@ -156,4 +159,30 @@ Object* get_prototype_from_constructor(GlobalObject& global_object, Function con
     return &prototype.as_object();
 }
 
+// 19.2.1.1 PerformEval ( x, callerRealm, strictCaller, direct )
+Value perform_eval(VM& vm, Value x, GlobalObject& caller_realm, CallerMode strict_caller, EvalMode direct)
+{
+    assert(direct == EvalMode::Direct || strict_caller == CallerMode::NonStrict);
+    if (!x.is_string())
+        return x;
+
+    auto& code_string = x.as_string();
+    Parser parser { Lexer { code_string.string() } };
+    auto program = parser.parse_program();
+
+    if (parser.has_errors()) {
+        auto& error = parser.errors()[0];
+        vm.throw_exception<SyntaxError>(caller_realm, error.to_string());
+        return {};
+    }
+
+    auto& interpreter = vm.interpreter();
+    if (direct == EvalMode::Direct) {
+        return interpreter.execute_statement(caller_realm, program).value_or(js_undefined());
+    }
```

Comment:
We typically omit braces for single-line if-statements
```suggestion
    if (direct == EvalMode::Direct)
        return interpreter.execute_statement(caller_realm, program).value_or(js_undefined());
```